### PR TITLE
ICDS Dashboard: Service delivery dashboard initial changes

### PR DESCRIPTION
**Pull request by saiprakash-v**
Trello: https://trello.com/c/8iBlwmIN
##### SUMMARY

This PR includes some refactoring and other necessary changes for SDD on mobile dashboard
##### FEATURE FLAG

Service Delivery Dashboard is added as a menu item in mobile nav-bar under feature flag. Although Routes are added in this pr, there is no UI of mobile SDD included here.

### CODE REVIEW

File: custom/icds_reports/templates/icds_reports/icds_app/mobile/navigation.directive.html [hunk 1]
```diff
@@ -28,8 +28,12 @@
             </div>
         </div>
         <hr>
+        <div class="menu-options navigation-font mb-0" ng-show="$ctrl.haveAccessToFeatures" ng-class="{'active-mobile-link': $location.path().indexOf('/service_delivery_dashboard') !== -1}">
+            <i class="fa fa-tasks mr-3" aria-hidden="true"></i>
+            <a  href="{$ goToStep('service_delivery_dashboard/', $location.search()) $}" class="sub-links" ng-click="closeMenu()">Service Delivery Dashboard</a>
+        </div>
         <div class="menu-options navigation-font mb-0" ng-class="{'active-mobile-link': $location.path().indexOf('/awc_reports') !== -1}">
-            <img src="{% static 'icds_reports/mobile/images/awc-reports.png' %}" class="mr-3">
```
Comment (czue) [inline custom/icds_reports/templates/icds_reports/icds_app/mobile/navigation.directive.html]: why was this removed?

File: custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js [hunk 2]
```diff
@@ -76,38 +76,172 @@ function ServiceDeliveryDashboardController($rootScope, $scope, $http, $location
...
+    vm.sddTableData = {
+        'pw_lw_children': {
+            'non-awc' : [
+                {
+                    'mData' : 'num_launched_awcs',
+                    'heading' : 'Number of AWCs launched',
+                    'tooltipValue' : 'Total Number of Anganwadi Centers launched in the selected location.',
+                    'columnValueType' : 'raw',
+                    'columnValueIndicator' : 'num_launched_awcs',
+                    'stylingClass' : 'medium-col'
```
Comment (czue) [inline custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js]: nit: this appears to be the same on every item - should we just remove it and always assume the value or you expect it to change for some?

File: custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js [hunk 3]
```diff
@@ -76,38 +76,172 @@ function ServiceDeliveryDashboardController($rootScope, $scope, $http, $location
...
+            vm.sddTableData['children'];
+    };
+
+    vm.isPwLwChildrenTab = function () {
+        return ($location.path().indexOf('/pw_lw_children') !== -1);
+    };
+
+    vm.isAwcDataShown = function () {
+        // if sector level is selected --> selectedLocationLevel is 3
+        return (parseInt($location.search()['selectedLocationLevel'], 10) === 3);
```
Comment (czue) [inline custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js]: what does the `10` do here?

File: custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js [hunk 4]
```diff
@@ -76,38 +76,172 @@ function ServiceDeliveryDashboardController($rootScope, $scope, $http, $location
...
-                ]);
-            }
-        } else {
-            vm.dtColumns = vm.dtColumns.concat([
-                DTColumnBuilder.newColumn('sn').withTitle(renderSupplementaryNutritionTooltip()).renderWith(renderCellValue('percentage','supNutrition')).withClass('medium-col'),
-                DTColumnBuilder.newColumn('pse').withTitle(renderPreSchoolEducationTooltip()).renderWith(renderCellValue('percentage','pse')).withClass('medium-col'),
-                DTColumnBuilder.newColumn('gm').withTitle(renderGrowthMonitoring36Tooltip()).renderWith(renderCellValue('percentage','gm36')).withClass('medium-col'),
-            ]);
-        }
+        vm.dtColumns = vm.dtColumns.concat(vm.buildDataTable(vm.isPwLwChildrenTab(), vm.isAwcDataShown()));
```
Comment (czue) [inline custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js]: nice, this looks like a good change :)

**Review by czue — APPROVED** [review]
This looks great, thanks. It's a fair amount of code that could touch the web dashboard which always makes me a little nervous - though it looks like everything was pretty mechanical. Do you have any recommendations on what level of testing we should try to do before pushing this to production? One thought is we could merge and request a sanity QA on the india test server first.

**Comment by czue** [comment]
(just putting a "don't merge" on until we decide on a plan for testing/rollout)

File: custom/icds_reports/templates/icds_reports/icds_app/mobile/navigation.directive.html (hunk 1 above)
Comment (saiprakash-v) [reply custom/icds_reports/templates/icds_reports/icds_app/mobile/navigation.directive.html]: Saket asked me to change icon, So i removed this and added the other one

File: custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js (hunk 2 above)
Comment (saiprakash-v) [reply custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js]: Yeah!! I will remove it. As of now, it is pretty standard that this styling is followed for all the datatable columns

File: custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js (hunk 3 above)
Comment (saiprakash-v) [reply custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js]: its a radix/base for parsing string to integer. (eg: to convert it to binary it will be 2, for decimal it will be 10)

File: custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js (hunk 4 above)
Comment (saiprakash-v) [reply custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js]: > This looks great, thanks. It's a fair amount of code that could touch the web dashboard which always makes me a little nervous - though it looks like everything was pretty mechanical. Do you have any recommendations on what level of testing we should try to do before pushing this to production? One thought is we could merge and request a sanity QA on the india test server first.

I'm positive that it will not break any current dashboard functionalities. Yeah, you can deploy it in india server first for testing

Comment (czue) [reply custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js]: Strong words! Ok sounds like a plan. :muscle:

**Comment by czue** [comment]
kicked the build
